src/protocols/ab/ab_common.c

```diff
@@ -380,10 +379,26 @@ plc_tag_p ab_tag_create(attr attribs)
     /* get the element count, default to 1 if missing. */
     tag->elem_count = attr_get_int(attribs,"elem_count", 1);
 
-    /* we still need size on non Logix-class PLCs */
-    if(tag->plc_type != AB_PLC_LGX
-            && tag->plc_type != AB_PLC_MLGX800
-            && tag->plc_type != AB_PLC_OMRON_NJNX) {
+
+    switch(tag->plc_type) {
+    case AB_PLC_OMRON_NJNX:
+        if (tag->elem_count != 1) {
+            tag->elem_count = 1;
+            pdebug(DEBUG_WARN,"Attribute elem_count should be 1!");
+        }
+
+        /* from here is the same as a AB_PLC_MLGX800. */
+
```

Comment:
So this falls through?   I think the compiler will complain about that.  I will need to check.  Generally I try to have fall through only on cases where each `case` branch is identical.   Otherwise I try to break out each `case` branch as completely independent code.

Reply:
> Thanks!
> 
> I think it might be time to pull the Omron stuff out into its own set of code, even though that will make a lot of duplication. It is a weird PLC, isn't it?
> 
> I will try to refactor the Omron code out this weekend and see if can cut down on the duplicate and complicated code paths. For instance, I suspect that the Omron cannot handle unconnected messaging (UCMM). That is a big part of eip_cip.c.
> 
> What happens if you do send an element count of something other than 1?

A request error, don't have access to the error code ATM.

I agree that it will be better to give OMROM his own code.